integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java

```diff
@@ -280,9 +297,13 @@ private void createBlockFile(String filename, int len) throws IOException, Inval
     out.close();
   }
 
-  // Sleeps for a duration so that the worker heartbeat to master can be processed
-  private void waitForHeartbeat() {
-    CommonUtils
-        .sleepMs(mWorkerTachyonConf.getInt(Constants.WORKER_BLOCK_HEARTBEAT_INTERVAL_MS) * 3);
+  // Waits for a worker heartbeat to master to be processed
+  private void waitForHeartbeat() throws InterruptedException {
```

Comment:
`HeartbeatScheduler.await` first checks `sTimers.containsKey(name)`, then only calls `sCondition.await` if the heartbeat isn't ready for scheduling. If the heartbeat has already been waited for once, a second await will return immediately. This is nice because we don't have to worry about calling `HeartbeatScheduler.await` too many times.